### ISSUE TITLE
Don't throw `PossiblyFalseArgument` on pass `false|list<string>` to `in_array`

### ISSUE THREAD

**Issue by vjik** [vimeo/psalm#9764]
https://psalm.dev/r/e229d24bd7

Var is `false|list<string>`, but `PossiblyFalseArgument` into `in_array` do not throw.

Expected:

```
[PossiblyFalseArgument](https://psalm.dev/104) - 7:26 - Argument 2 of in_array cannot be false, possibly array<array-key, mixed> value expected
```

**Comment by orklah** [comment]
It will throw if you put this config to false: https://psalm.dev/docs/running_psalm/configuration/#ignoreinternalfunctionfalsereturn

It does not by default because of the bad habit PHP had to return exotic false/null when given invalid input before PHP 8 (so Psalm would ignore them to preven bothering you to check for every possible exotic value). I'd be open to change the default value for this config in order to report them by default now that PHP 8 cleaned up everything

**Comment by weirdan** [comment]
Fixed in #10211
